Growth - Auras, Effects, and Behaviors

parkerhams opened the pull request:
Overall, it is not behaving how intended. The Tree model - the RedFirTree prefab - is not scaling the model, but it *is* scaling the trigger area around the tree. As such, I made a debug version of the prefab with a simple sphere - and the sphere does scale. 

I think it might have something to do with the ways the meshes are working in the tree model itself? Since the sphere object has a mesh renderer and collider by default, and the tree model does on the children objects, it makes me think it has something to do with that. 

But please let me know ways you think I could make it better!!!

Commit:
Created sphere trigger around player that changes material color values upon hitting trigger area of the Flora tag! Yay!

dantogno committed:
Merge branch 'master' into player-aura

parkerhams committed:
Merge branch 'master' into player-aura

parkerhams committed:
Renamed the scene to PlayerAuraScene for better readability. Also updated the player prefab with the Aura script wo that we can see the debug happening!

parkerhams committed:
Changed commented code to tooltips and made things easier to read by getting rid of comment code, re-organizing - this script will be activated once the player's aura is detected

parkerhams committed:
The capsule is scaling, but downwards for some reason. Successfully it does not begin scaling down until the player corsses the trigger. Hoowever, the tree is not growing correctly quite yet.

parkerhams committed:
new editor values - increased the growth speed and decreased the time

parkerhams committed:
added sound effect to play when the scaling is happening, added call in script. scaling is still not being triggered in the desired way.

parkerhams committed:
For some reason, the tree is not growing - and I think it might be because of the repeatable boolean vlaue. Let's take a look at this next - as well as the spawnedFoliage lists, which are confusing even me. TIme to clean that portion up!

parkerhams committed:
Added particles around player that play when in the trigger area for growth

parkerhams committed:
So it is actively changing the scale in the editor to what I set it to, however it is not changing size visually in the game view. It stays the same size for some reason, perhpas it's the tree model I'm using?

parkerhams committed:
Emits particles once it reaches max scale

parkerhams committed:
Still does not scale the actual model - thus far, have tried getting rid of the repeatablle variable to be sure that the conditions are being met, but still is not scaling the model itself. I still see it scaling in the inspector

parkerhams committed:
tried making new version of the tree with a new empty that had childed the tree prefab from the imported assets as opposed to changing the actual asset itself to have these scripts; still did not work even when tagged as flora. Odd.

parkerhams committed:
goodness gracious I activated the GrowthTrigger object and this boi scales but the audio plays several several times over itself and the particle effect do not stop like they did previously

parkerhams committed:
CHanged OnTriggerStay to OnTriggerEnter - that was why the effects/sounds kept playing

parkerhams committed:
Took out any and all things related to foliage and put it in its own script SEPARATE from the scaleLerper, which only scales the object. having all of the code around foliage in there, especially when it didn't work, got confusing - better to try decoupling them. Now, ScaleLerper script **only** focuses on scaling the obejct. I do not use the FoliageGrowth script, YET.

parkerhams committed:
So for some reason, the assets I'm using are not working properly with scaling. I've attached the components to a basic sphere object, and the sphere scales. I believe it may have something to do with the mesh renderer? I also added a sound effect play once it reaches its maxScale. However, it's shrinking down now. Will make a PR to get feedback and suggestions.

parkerhams committed:
Tried messing with the tree's colliders, but still not behaving how intended.

parkerhams committed:
tried adjusting the tags

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FoliageGrowth : MonoBehaviour
```
Add a summary for the whole class with the <summary> tag  (hotkey: three forward slashes in visual studio)  https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/summary

parkerhams replied:
Added! Let me know if it makes sense - I can always make it more detailed or condense it to be readable.

dantogno replied:
Ok. I get it and I like the idea, but I think we should focus first on the core feature. Don't delete this or anything, I just want to put it on pause.

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+    [Tooltip("base size for any trees or foliage that will scale")]
+    Vector3 minScale;
+
+    [Tooltip("max scale meant to be controlled in editor - biggest size it can get")]
```
"The biggest size it can get" is sufficient, I'd delete the other words in the tip text.

parkerhams replied:
Changed it to that!

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+    [SerializeField]
+    Vector3 maxScale;
+
+    [Tooltip("checking scales later to then spawn more foliage around tree")]
```
I don't know what this means. Clarify.

parkerhams replied:
"location of spawn areas for more thickness and brush"

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+    [SerializeField]
+    Transform foliageSpawn;
+
+    [Tooltip("how quickly it grows")]
```
How about "growthRate?" Speed implies traveling.

parkerhams replied:
renamed to growthRate!

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+    [SerializeField]
+    private float speed = 2f;
+
+    [Tooltip("can it also shrink down - Meant for if player has not completely grown the area")]
```
I understand the concept from the tip, but I think the name "repeatable" could be clarified. how about `canShrink`?

parkerhams replied:
canShrink makes a lot more sense

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+    bool repeatable = true;
+
+    [SerializeField]
+    bool isGrowingFoliage;
```
Clarify

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+    bool isGrowingFoliage;
+
+    [SerializeField]
+    static float foliageGrowth = 0.0f;
```
clarify

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+    [SerializeField]
+    List<GameObject> foliage = new List<GameObject>();
+
+    List<GameObject> spawnedFoliage = new List<GameObject>();
```
clarify

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+    public void TriggerGrowth()
+    {
+        // .. and increase the foliageGrowth interpolater
+        foliageGrowth += 0.5f * Time.deltaTime;
```
`.5f` is a magic number here. Make it a variable with a name so I understand what it is.

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+            isGrowingFoliage = true;
+            if (isGrowingFoliage)
+            {
+                Vector3 temp9 = new Vector3(Mathf.Lerp(minScale.y, maxScale.y, foliageGrowth), 0);
```
what the heck is `temp9`? come up with a more descriptive name.

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+                Vector3 temp9 = new Vector3(Mathf.Lerp(minScale.y, maxScale.y, foliageGrowth), 0);
+                temp9 = foliageSpawn.transform.position;
+
+                //FoliageSpawnLerp(foliageSpawn, maxScale, duration);
```
remove commented out code

dantogno commented on Assets/Scripts/FoliageGrowth.cs:
```diff
@@ -0,0 +1,101 @@
+
+    }
+
+    public IEnumerator FoliageSpawnLerp(Transform location, Vector3 size, float period)
```
I'm not sure you are using this currently, but before we delete it, I want to understand more about the intent of the class itself.

dantogno commented on Assets/Scripts/PlayerAuraEmit.cs:
```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+//The aura trigger surrounding the player that calls for plants to grow.
```
convert this comment into a summary tag

dantogno commented on Assets/Scripts/PlayerAuraEmit.cs:
```diff
@@ -0,0 +1,70 @@
+
+    [Tooltip("The Color to show aura has been activated")]
+    [SerializeField]
+    Color activatedAuraColor;
```
These are great names btw 👏

dantogno commented on Assets/Scripts/PlayerAuraEmit.cs:
```diff
@@ -0,0 +1,70 @@
+    // Start is called before the first frame update
+    void Start()
+    {
+        startLerpTime = Time.time;
```
what is the intent of `startLerpTime`?

dantogno commented on Assets/Scripts/PlayerAuraEmit.cs:
```diff
@@ -0,0 +1,70 @@
+
+    void ActivateAuraColorLerp()
+    {
+        float t = (Time.time - startLerpTime) * lerpSpeed;
```
Once I know what `startLerpTime` is meant to be I may have more comments for this.

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
-    [SerializeField]
-    bool repeatable;
+
+    [Tooltip("scalableObject is a reference to the location where the foliage should spawn")]
```
I don't understand the connection between the tip and the name. Clarify.

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
-    static float foliageGrowth = 0.0f;
-    
-    //foliage to add more details like grass or flowers around these larger scaled objects
+    private float speed = 2f;
```
`growthRate` is a better name.

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
-    List<GameObject> foliage = new List<GameObject>();
-    List<GameObject> spawnedFoliage = new List<GameObject>();
-    //scalableObject is a reference to the location where the foliage should spawn
+    private float duration = 5f;
```
wouldn't changing the `growthRate` also change the `duration`? How can the user set them both independently?

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
+
     [SerializeField]
-    GameObject scalableObject;
+    AudioSource auraAudio;
```
Use a single audio source with multiple clips (modify the `AudioSource.clip` property). In other words, change the type of this to `AudioClip.` Depending on what game object the `AudioSource` component is on, you can initialize it with `GetComponent` (since there will only be one).

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
-    //how long to process of growing takes place
-    public float duration = 5f;
+    [SerializeField]
+    AudioSource completedGrowingAudio;
```
make this an `AudioClip`

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
+    AudioSource completedGrowingAudio;
 
-    
+    [SerializeField]
```
add a quick tooltip

dantogno replied:
I think you could rename `doneGrowingParticles` to something like `growthCompletedParticles`, because it almost sounds like "I'm done growing particles" which is not the intent:
```
if (doneGrowingParticles)
  // Do the next thing now that the particles are done growing.
```

I assume the intent is that these are particles meant to play with the growth process is done.

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
-            //lerp scale down for fluctuation?
-            yield return RepeatLerp(maxScale, minScale, duration);
-            
+            StartCoroutine("Detect");
```
I prefer passing in the actual method as a parameter as opposed to a string. It enforces typo-free code at compile time

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
 
-        // .. and increase the foliageGrowth interpolater
-        foliageGrowth += 0.5f * Time.deltaTime;
+    public IEnumerator Detect()
```
definitely add a summary to this. Also, does it need to be public?

dantogno replied:
Once I understand the intent more clearly, you probably want to rename this method as well.

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
 public class ScaleLerper : MonoBehaviour
 {
-    //base size for any trees or foliage that will scale
+    [Tooltip("base size for any trees or foliage that will scale")]
```
if this is not a serialized field, don't have a tooltip.

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
+    public IEnumerator Detect()
+    {
+        //smallest scale is whatever it is set to in world space
+        minScale = transform.localScale;
```
maybe you should initialize this in Start

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -4,75 +4,58 @@
-        }
     }
 
     //takes in two vector 3s - start scale and max scale - and how quickly they lerp as time
```
make this a proper summary

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -83,55 +66,24 @@ public IEnumerator RepeatLerp(Vector3 a, Vector3 b, float time)
         float rate = (1.0f / time) * speed;
```
what is i? what is this rate formula meant to do?

parkerhams replied:
renamed rate to be scaleRate, renamed i to be iterateLerpRate. 

 float scaleRate = 0.0f;
float iterateLerpRate = (1.0f / time) * speed;

This is meant to take the speed set in the editor and actually scale it upwards by the iterateLerpRate and a factor of Time.deltaTime -- scaleRate += Time.deltaTime * iterateLerpRate;

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -83,55 +66,24 @@ public IEnumerator RepeatLerp(Vector3 a, Vector3 b, float time)
         float rate = (1.0f / time) * speed;
 
         //this while loop was made while following along with Resistance Code! Not my while loop
```
what the heck is Resistance Code? 😄

parkerhams replied:
Just credited the tutorial source 💃

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -83,55 +66,24 @@ public IEnumerator RepeatLerp(Vector3 a, Vector3 b, float time)
             i += Time.deltaTime * rate;
             //changing the world scale of the object to whatever it is on those three conditions
-            transform.localScale = Vector3.Lerp(a, b, i);
+            //transform.localScale = Vector3.Lerp(a, b, i);
```
remove commented out code

parkerhams replied:
removed 👍

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -83,55 +66,24 @@ public IEnumerator RepeatLerp(Vector3 a, Vector3 b, float time)
         {
             //this is mine though hehe
-            //yeet
             i += Time.deltaTime * rate;
```
what's this meant to do

parkerhams replied:
This is meant to take the speed set in the editor and actually scale it upwards by the iterateLerpRate and a factor of Time.deltaTime -- scaleRate += Time.deltaTime * iterateLerpRate;

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -83,55 +66,24 @@ public IEnumerator RepeatLerp(Vector3 a, Vector3 b, float time)
-            }
-        }
+        doneGrowingParticles.Play();
+        yield return new WaitForSeconds(2f);
```
magic number `2`. Make it a variable with a name so I know what it is for.

parkerhams replied:
added the particleDuration float variable

dantogno reviewed (changes requested):
@parkerhams Nice progress. Work on replying or committing changes to answer all of my questions. My goal is to understand your intent clearly, then we can start focusing on functionality. Keep up the good work!

parkerhams committed:
made code changes according to feedback on PR 17 - renamed variables to make more sense contextually

parkerhams committed:
Added summary for FoliageGrowth script

parkerhams committed:
renamed temp varibale i to scaleRate

parkerhams committed:
More detailed summary

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -62,25 +65,23 @@ public IEnumerator Detect()
-        float i = 0.0f;
-        float rate = (1.0f / time) * speed;
+        float scaleRate = 0.0f;
+        float iterateLerpRate = (1.0f / time) * speed;
```
Better add a summary to iterateLerpRate becaues I'm not totally clear on it  😉

parkerhams replied:
I hope this might make some more sense of what it's doing - please let me know if there's another way we could do this that might be more clear of efficient!

"This Enumerator is using scaleLerp to tell the gameObject where to start - and when we reach
  the iterateLerp variable, in this case 1 divided by the passed in time variable multipled by speed,
  we are telling the game objects to stop scaling. While we scale upwards, we scale while we are 
  between the values of scaleRate and iterateLerpRate. We start at a value of 0 here, and once the 
 value reaches the iterateLerpRate value, we stop scaling upwards."

dantogno replied:
Yeah this is still not clear enough for me to follow and understand easily. Take another pass at naming these things and writing clear summaries for anything where the name is not crystal clear. 

- For example, a normal person is not going to really know what an `iterateLerpRate` is. So it needs a better name and / or a summary. 
- Why would you scale between the values of scaleRate and iterateLerpRate? Wouldn't I scale between the values of the min or starting scale  and the maximum or ending scale?

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -32,6 +32,9 @@ public class ScaleLerper : MonoBehaviour
     [SerializeField]
```
Add a summary tag for this class (I guess I can't comment at the top of the class because there are no changes there in this PR)

parkerhams replied:
Done!

dantogno reviewed (changes requested):
@parkerhams A few more things to clarify please!

parkerhams committed:
made changes to variable names and added summaries in our ScaleLerper class to make more sense of things

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -64,7 +71,13 @@ public IEnumerator Detect()
     public IEnumerator RepeatLerp(Vector3 a, Vector3 b, float time)
     {
-        //rate of growth
+        /// <summary>
```
summary tags have to go on the line just before the thing they are summarizing. So not inside of the method, but just before it.

dantogno replied:
I think there may be some confusion by the fact that I asked for two summaries here, one for the whole `RepeatLerp` method, and one for the field `iterateLerpRate`.

parkerhams replied:
Renamed stuff to be more honest about what it does - so RepeatLerp is now **ScaleLerp** because that's what it's doing, scaleRate is now **scaleValue** and iterateLerpRate was not a good name so now it is **increaseRate**. 

Here is the summary for ScaleLerp() coroutine:
 ---SCALELERP SUMMARY---
    /// In the Detect coroutine, we return RepeatLerp once the player enters the trigger area.
    /// When Detect is called, we set the minScale to whatever the growable GameObject's local scale is
    /// in the world space. We then call the couroutine RepeatLerp, which takes in two vector 3's and a float variable.
    /// We pass in minScale and maxScale and the duration, all serialized variables at the top, into our ScaleLerp coroutine.
    /// The ScaleLerp uses two temporary variables - scaleValue, which is 0 and is assigned as our starting point, and 
    /// increaseRate, which is used to see when we tell the GameObject to stop scaling. Our increaseRate value tells
    /// the scaleValue value how much to increase by until it gets to 1 over same length of time that it takes our scaleableObject to 
    /// reach maxScale. In short, while our scaleValue is greater than 0 but less than 1, 
    /// then we tell the object to scale to our a, b, and time values - 
    /// which are actually our minScale, maxScale, and durtion values. The increaseRate value is how we increment
    /// scaleValue to 1 - and when it reaches 1, then we stop scaling. We scale the GameObject in the while loop to by setting the
    /// scaleableObject's localScale to a Vector 3 lerp and passing in the a(minScale), b(maxScale), and time (duration) values.

Here is the summary for the scaleValue and increaseRate temporary variables:
 ----scaleValue and increaseRate summary----
        /// scaleValue is a temporary value used to say that when the scaleableObject is at its minScale, it is at 0, 
        /// and while the scaleableObject grows to its maxScale over the duration variable value, then we scale upwards 
        /// as long as scaleValue is less then 1 in a while loop. We increase scaleValue to 1 by adding Time.deltaTime multipled by
        /// the increaseRate value. increaseRate tells scaleValue to increase to 1 at the same rate that our scaleableObject 
        /// increases from minScale to maxScale - it accounts for the speed at which we've told the scaleable object to grow as well
        /// as the duration of growth through our temporary time variable, which takes in our serialized duration variable 
        /// when we call RepeatLerp() in our Detect() coroutine. While scaleRate is less than 1, scale the gameobject. 
        /// When scaleValue reaches 1 by taking into account the speed and time values, 
        /// then stop growing because that means we've reached our maxScale.

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -2,6 +2,13 @@
 using UnityEngine;
 
+/// <summary>
+/// The scale leraper class is meant to scale a game object, in this case trees, upwards to a maxScale
```
Typo "leaper"

parkerhams replied:
Fixed - thanks for catching that, my bad

dantogno commented on Assets/Scripts/ScaleLerper.cs:
```diff
@@ -2,6 +2,13 @@
+/// The scale leraper class is meant to scale a game object, in this case trees, upwards to a maxScale
+/// while the player's game object is in the trigger area of the tree.
+/// The trees are referred to as a GameObject scalableObject, which grows to the maxScale variable.
+/// The trees will grow ar a rate of the speed variable over a duration of the duration variable. 
```
typo "ar"

dantogno replied:
Mathematically, I don't think we can have independent duration, max, and rate variables. If we adjust one, the others are linked and will be adjusted as well. For example, if I am going from 0 to 10 at 1 per second, it's going to take me 10 seconds. I can't have the user change that 10 second duration without changing either the min, max, or rate as well. Does that make sense?

parkerhams replied:
fixed typo

That makes sense; the player themselves doesn't have the ability to manipulate these values but mathematically if we're using these I definitely want it to be the clearest it can be. If it's better to use one value or change what's here, let me know!

dantogno reviewed (changes requested):
The main issue remaining is that I really don't know what a RepeatLerp or an IterateLerpRate is, and the summaries are not helping me see the light just yet. It's very important to be able to express the intent of this clearly. 

If we can't express the meaning and intent of these clearly, it is a good indication that we need to rethink the approach to how we're implementing the feature.

parkerhams committed:
Added a summary for the ScaleLerp coroutine - recently renamed from RepatLerp so that is describes what we're doing more clearly - and added another new summary for the scaleValue (renamed from scaleRate) and the increaseRate (renamed from the iterateLerpRate to make more sense). Also fixed some typos. Renamed variables to be more clear about what they're used for.

parkerhams committed:
Shortened comments for temp variables!

parkerhams committed:
Merge branch 'master' into player-aura

parkerhams committed:
Added Cursor Hide to Player Prefab

parkerhams committed:
Hot damn it works

parkerhams committed:
Made adjustments to the speed value, also added the Shrink coroutine that does the same thing as the Grow coroutine right now expect that it takes in the minScale scale as a parameter, also deleted extra code that didn't help us from before

parkerhams committed:
Tried to stop coroutines grow when the player left the trigger area and stopcoroutine shrink when player entered trigger area - end conditions not being met, also still scaling downward incredibly slow

parkerhams committed:
Trying to switch around some logic flow to see if I can actually get the cylinder to reach the scale we set - it always stops at .0005 before it reaches whatever I've set maxScale at i nthe editor, so it's not technically reaching that value.

parkerhams committed:
Added if statement checking the localScale x y z values compared to the maxScale x y z minus a new vector3 threshold - if the localScale is within that maxScale - doneGrowingThreshold value

parkerhams committed:
Scales upwards now! Unity doesn't crash

parkerhams committed:
Now it just automatically goes to the maxScale based on the if statement I have set here

parkerhams committed:
Still checking how I can use the growthSpeed and set the value to maxScale without just setting and skipping the actual growth time

parkerhams committed:
so... when player touches the cylinder, Unity freezes and crashes.

parkerhams committed:
Scaling upwards works! Big yeet. HOwever, if youleave the area and let it shrink, it does not shrink down to its minScale - and if you re-enter it, it does not scale upwards to maxScale and scales a lot more slowly as if fighting a curve of some kind.

parkerhams committed:
David saved my ass too

parkerhams committed:
Tried making Shrink coroutine use time elements more - it stops scaling, but it is not shrinking

parkerhams committed:
Changed audiosource sound settings like pitch and volume

parkerhams committed:
Added music folder and song

parkerhams committed:
added music

parkerhams committed:
Assigned variables in editor

parkerhams committed:
StopAllCoroutines but it grows really slowly

dantogno committed:
switched from coroutine to fixedupdate, changed particle duration in editor for auara test scene.

becaues we aren't using the coroutine to stop the particles, we make the not loop and adjust duration.

parkerhams committed:
Reverse engineered the scaling elements in FixedUpdate to try and shrink - it doesn't shrink, but it stops growing! Yee haw

parkerhams committed:
When player exits, it snaps back to minscale - commented out the compareTag case in the if statement in OnTriggerExit, need to look at that

parkerhams committed:
isCloseEnough when we are trying to Shrink is what's stopping it from shrinking - when I comment the if statement checking if  isCloseEnough to min, it either snaps or does not grow - with that commented it does shrink properly!

parkerhams committed:
Changed line 144 from distanceFromMin to be instead distanceFromMax

parkerhams committed:
Merge branch 'master' into player-aura

parkerhams committed:
updating branch from master part 1 lmao
